### PR TITLE
Travis optimization

### PR DESCRIPTION
**Pull request by abooij**
I have done a bunch of experimentation with the travis setup. So I now indeed propose to start building `HomologicalAlgebra` sooner (closing #795). This change takes about 5-10 minutes off of the time from commit to CI completion.

I have tried the new container-based instances. Although they boot faster, they also seem to process noticably slower. So overall it is faster to use full virtualization (the current choice).

I made Travis install the required packages using the `apt` add-on, instead of the `apt-get install` commands. This is not necessarily faster, but it's more elegant, and enables switching to container-based instances.

The next obvious optimization steps would be:

- Optimize / break up `HomologicalAlgebra` (longest component at 19-23 minutes)
- Cache the coq build (this takes 2 minutes for every build that is done)

I have attempted to cache the coq build, but I didn't manage to write something that I felt was reliable. So this PR excludes that.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,25 +6,32 @@ sudo: 		required
...
-before_install:
-  - sudo apt-get update -qq
-install:
-  - sudo apt-get install -y ocaml ocaml-nox ocaml-native-compilers ocaml-findlib
-  - sudo apt-get install -y camlp5 camlp4-extra time libgtk2.0 libgtksourceview2.0
-  - sudo apt-get install -y liblablgtk-extras-ocaml-dev
-  # get etags from emacs:
-  - sudo apt-get install -y emacs
+cache:
+  apt: true
```
Comment (DanGrayson) [inline .travis.yml]: What does this line do?

Comment (abooij) [reply .travis.yml]: [nothing](https://github.com/travis-ci/travis-ci/issues/5876), apparently.

Comment (DanGrayson) [reply .travis.yml]: How about removing lines 9-10, in that case?

Comment (abooij) [reply .travis.yml]: Done!

**Comment by DanGrayson** [comment]
lgtm

**Comment by benediktahrens** [comment]
lgtm

**Comment by benediktahrens** [comment]
This PR will be in conflict with #785 .

**Comment by palmskog** [comment]
@benediktahrens if you or @DanGrayson merge this PR before #785, I will update #785 to incorporate the changes here. They are quite orthogonal, although some syntax changes overlap.

**Pull request merged**